src/Core/ViewHelper/AbstractConditionViewHelper.php

```diff
@@ -90,59 +90,65 @@ public static function verdict(array $arguments, RenderingContextInterface $rend
      */
     protected function renderThenChild()
     {
-        // Prefer "then" ViewHelper argument if present
-        if ($this->hasArgument('then')) {
-            return $this->arguments['then'];
-        }
-
-        // Closure might be present if ViewHelper is called from a cached template
+        // In cached templates, a closure is defined if any variant of "then" has been specified
         if ($this->thenClosure !== null) {
-            return ($this->thenClosure)() ?? '';
+            return ($this->thenClosure)();
         }
 
         // The following code can only be evaluated for uncached templates where the node structure
         // is still available. If it's not, it has already been executed during compilation and we can
         // assume that the condition wasn't met
         if (!$this->viewHelperNode instanceof ViewHelperNode) {
-            return '';
+            return null;
         }
 
+        // node arguments are used because the ViewHelper arguments are already merged with default
+        // values, which makes it impossible to differentiate between "argument is not defined" and
+        // "argument returns null"
+        $nodeArguments = $this->viewHelperNode->getArguments();
+
+        // Prefer "then" ViewHelper argument if present
+        if (array_key_exists('then', $nodeArguments)) {
+            return $this->arguments['then'];
+        }
+
+        // Search for f:then ViewHelper and identify possible f:else to decide how the tag's children should
+        // be interpreted afterwards
         $elseViewHelperEncountered = false;
         foreach ($this->viewHelperNode->getChildNodes() as $childNode) {
             if ($childNode instanceof ViewHelperNode
-                && substr($childNode->getViewHelperClassName(), -14) === 'ThenViewHelper') {
-                $data = $childNode->evaluate($this->renderingContext);
+                && str_ends_with($childNode->getViewHelperClassName(), 'ThenViewHelper')) {
```

Comment:
That looks error prone to me - but as this not really a change i would not discuss this here. But what is if i have a "HereAndThenViewhelper" that does something total unrelated? Relying on the ending of the viewhelper class does not look very stable to me. Maybe there should be an interface to check for?
Should we create a ticket for this?

Reply:
Yes, please do. I only touched this to get rid of that substr() call.

Reply:
Done 1053